src/Microsoft.SymbolStore/SymbolStoreKey.cs

```diff
@@ -104,7 +107,7 @@ public static bool IsKeyValid(string index)
                     if (char.IsLetterOrDigit(c)) {
                         continue;
                     }
-                    if (c == '_' || c == '-' || c == '.' || c == '%') {
+                    if (!s_invalidChars.Contains(c)) {
```

Comment:
Not sure this will work as what we thought. The dictionary returns a bigger set on windows than on Linux. Linux files are allowed to have such characters, and we index our symbols on a Windows leg. Not sure if the symbol server can store this files, but the key creation should probably not fail in such case.  I consider what you are doing here an improvement nonetheless as at least this now allows to index all unicode named files.

Form the documentation:
> The array returned from this method is not guaranteed to contain the complete set of characters that are invalid in file and directory names. The full set of invalid characters can vary by file system. For example, on Windows-based desktop platforms, invalid path characters might include ASCII/Unicode characters 1 through 31, as well as quote ("), less than (<), greater than (>), pipe (|), backspace (\b), null (\0) and tab (\t).

Reply:
The difference between Windows/Linux/MacOS might be what  we want because this function is only used by the Cache and DirectorySymbolStores which are writing files based on parts of the index to disk. If IsKeyValid fails, the cache or directory entry will be skipped.